integrations/slack/webhooks/webhook.go

```diff
@@ -195,3 +234,46 @@ func (h handler) dispatchAsyncEventsToConnections(ctx context.Context, cids []sd
 		l.Debug("Event dispatched")
 	}
 }
+
+func (h handler) extractIDs(body []byte, wantContentType string, l *zap.Logger) (appID, teamID, enterpriseID string, err error) {
+	if wantContentType == "application/json" {
+		var cb events.Callback
+		if err := json.Unmarshal(body, &cb); err != nil {
+			l.Error("Failed to parse JSON for app/team IDs",
+				zap.Error(err),
+			)
+			return "", "", "", err
+		}
+		return cb.APIAppID, cb.TeamID, "", nil
+	}
+
+	// Handle form data
+	kv, err := url.ParseQuery(string(body))
+	if err != nil {
+		l.Error("Failed to parse URL-encoded form",
+			zap.ByteString("body", body),
+			zap.Error(err),
+		)
+		return "", "", "", err
+	}
+
+	// Check if this is an interaction payload
+	if payload := kv.Get("payload"); payload != "" {
+		var p BlockActionsPayload
+		if err := json.Unmarshal([]byte(payload), &p); err != nil {
+			l.Error("Failed to parse interaction payload",
+				zap.String("payload", payload),
+				zap.Error(err),
+			)
+			return "", "", "", err
+		}
+
+		if p.IsEnterpriseInstall && p.Enterprise != nil {
```

Comment:
Instead of 2 separate returns, use a helper variable which is "" by default, and return it either way,